test/code/smoke.cjs

```diff
@@ -27,38 +27,196 @@ test(
 
-            // Make HTML - move to subclass!
-            const ret = {
-                     docSetId,
-                     mainSequenceId: output.docSets[docSetId].documents[bookCode].mainSequence,
-                     headers: output.docSets[docSetId].documents[bookCode].headers,
-                     sequenceHtml: {},
-                 };
-            Object.keys(output.docSets[docSetId].documents[bookCode].sequences)
-                .forEach(seqId => {ret.sequenceHtml[seqId] = perf2html(output, seqId)});
-            // console.log(JSON.stringify(ret, null, 2));
-            
-        } catch (err) {
-            t.error(err);
-            console.log(err);
-        }
-    },
-);
+        t.test(
+            `fetchPerf should not fetch wrong bookCode`,
+            async t => {
+                t.plan(1);
+                try {
+                    const bookCode = "LU";
+                    await epitelete.fetchPerf(bookCode);
+                    t.fail("fetchPerf with bad bookCode should throw but didn't");
+                } catch (err) {
+                    t.pass("fetchPerf throws on bad bookCode");
+                }
+            }
+        )
+
+        t.test(
+            `fetchPerf() returns config output`,
+            async t => {
+                try {
+                    const bookCode = "LUK";
+                    const output = await epitelete.fetchPerf(bookCode);
+                    t.ok("docSets" in output);
+                    t.ok(docSetId in output.docSets);
+                    t.ok("documents" in output.docSets[docSetId]);
+                    t.ok(bookCode in output.docSets[docSetId].documents);
+                    const doc = output.docSets[docSetId].documents[bookCode];
+                    for (const k of ['headers', 'tags', 'sequences', 'mainSequence']) {
+                        t.ok(k in doc);
+                    }
+                    // Make HTML - move to subclass!
+                    /*
+                    const ret = {
+                        docSetId,
+                        mainSequenceId: output.docSets[docSetId].documents[bookCode].mainSequence,
+                        headers: output.docSets[docSetId].documents[bookCode].headers,
+                        sequenceHtml: {},
+                    };
+                    Object.keys(output.docSets[docSetId].documents[bookCode].sequences)
+                        .forEach(seqId => { ret.sequenceHtml[seqId] = perf2html(output, seqId) });
+                        */
+                    // console.log(JSON.stringify(ret, null, 2));
+                } catch (err) {
+                    t.error(err)
+                    console.log(err);
+                }
+                t.end()
+            }
+        )
+
+        t.test(
+            `fetchPerf() adds document to documents property`,
+            async t => {
+                t.plan(2);
+                const bookCode = "MRK";
+                t.notOk(bookCode in epitelete.documents);
```

Comment:
This is great! Check that the thing you want to happen hasn't already happened before you run the code!